platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -103,22 +103,6 @@ - (BOOL)intersectsOverlayBounds:(MGLCoordinateBounds)overlayBounds
     return MGLLatLngBoundsFromCoordinateBounds(_bounds).intersects(MGLLatLngBoundsFromCoordinateBounds(overlayBounds));
 }
 
-- (mbgl::AnnotationSegments)annotationSegments {
-    NSUInteger count = self.pointCount;
-    CLLocationCoordinate2D *coordinates = self.coordinates;
-    
-    mbgl::AnnotationSegment segment;
-    segment.reserve(count);
-    for (NSUInteger i = 0; i < count; i++) {
-        segment.push_back(MGLLatLngFromLocationCoordinate2D(coordinates[i]));
-    }
-    return { segment };
-}
-
-- (mbgl::ShapeAnnotation::Properties)shapeAnnotationPropertiesObjectWithDelegate:(__unused id <MGLMultiPointDelegate>)delegate {
```

Comment:
Currently, if you call `-[MGLMapView visibleFeaturesAtPoint:]` at the location of a multipoint (in geometry.hpp terms), you get an MGLMultiPoint object back. MGLMultiPoint conforms to MGLAnnotation, so you can add this MGLMultiPoint object to the map using `-[MGLMapView addAnnotations:]`. With these changes, MGLMapView will end up [calling `-shapeAnnotationObjectWithDelegate:`](https://github.com/mapbox/mapbox-gl-native/blob/98b9946f36df267fa04a04dcf2e6138ded0cb428/platform/ios/src/MGLMapView.mm#L2820) on an MGLMultiPoint that has no implementation of `-shapeAnnotationObjectWithDelegate:`, resulting in an Objective-C exception.

If it isn’t straightforward to create an `mbgl::ShapeAnnotation` representing a multipoint geometry at this time, I think it’d be reasonable to implement a `-shapeAnnotationObjectWithDelegate:` that asserts for now.